lib/SILOptimizer/Transforms/SpecializeOpaqueArchetypes.cpp

```diff
@@ -0,0 +1,466 @@
+#define DEBUG_TYPE "opaque-archetype-specializer"
+
+#include "swift/AST/Types.h"
+#include "swift/SIL/SILFunction.h"
+#include "swift/SIL/TypeSubstCloner.h"
+#include "swift/SIL/SILInstruction.h"
+#include "swift/SILOptimizer/PassManager/Transforms.h"
+#include "swift/SILOptimizer/Utils/CFG.h"
+
+
+using namespace swift;
+
+static Type substOpaqueTypesWithUnderlyingTypes(
+    Type ty, SILFunction *context) {
+  ReplaceOpaqueTypesWithUnderlyingTypes replacer(context);
```

Comment:
Please move ReplaceOpaqueTypesWithUnderlyingTypes into this file

Reply:
It is now also used by lib/SIL/SILFunctionType.cpp because substituted and unsubstituted are ABI compatible.

Reply:
Ah, ok